### PR TITLE
Add logout logic in 'Login' test case

### PR DESCRIPTION
**Pull request by AlekseyDidilov**
Login test case was implemented for checking ability to
login into user account (with valid credentials). After
checking that user logged into account, need to make
logout, for prevent conflicts with test cases where
logout isn't used.

Logout link located in drop-down menu in Header. For
ability to work with this link, 'Header' page was
implemented in oct/pages. #26

### CODE REVIEW

File: oct/tests/web/login.py [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
 from oct.tests import run_testcase
 from oct.pages.login import LoginPage
 from oct.pages.account import AccountPage
+from oct.pages.registration import RegisterAccountPage, RegistrationSuccessPage
 
 
 class Login(Testcase):
+    @setup
+    def registration(self, grid: str) -> None:
+        chrome: Remote = Chrome(grid)
```
Comment (extsoft) [inline oct/tests/web/login.py]: @AlekseyDidilov as per #26, you need to use API to create a user.

File: oct/tests/web/login.py [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+        registration = RegisterAccountPage(chrome)
+        registration.open()
+        registration.fill_personal_details(
+            "Alex", "Ivanov", "didilov.aleksey@gmail.com", "+38067212345"
```
Comment (extsoft) [inline oct/tests/web/login.py]: @AlekseyDidilov Looks like email and password has to be class variables as they are used in two methods.

Comment (AlekseyDidilov) [reply oct/tests/web/login.py]: Thanks, got it.

File: oct/tests/web/login.py (hunk 1 above)
Comment (AlekseyDidilov) [reply oct/tests/web/login.py]: @extsoft thanks, I rewrote setup section, and now user is being created via API.

File: oct/tests/web/login.py [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
 from oct.pages.login import LoginPage
 from oct.pages.account import AccountPage
+from oct.tests.api.registration import Registration
 
 
 class Login(Testcase):
+    @setup
+    def create_account(self) -> None:
+        with suppress(AssertionError):
+            Registration().test_registration()
```
Comment (extsoft) [inline oct/tests/web/login.py]: @AlekseyDidilov the ideal implementation is to extract an API call to a separate place and reuse it in both tests. Also, if user or password will be changed in the  `Registration` test, this test will fail. So, you have to define separate credentials here.

Comment (AlekseyDidilov) [reply oct/tests/web/login.py]: @extsoft Thank, done.

File: oct/tests/api/registration_pattern.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class RegistrationAPI:
+    def __init__(self, full_name: List[str], email: str, telephone: str, password: str):
+        self._full_name = full_name
+        self._email = email
+        self._telephone = telephone
+        self._password = password
+
+    def account_registration(self) -> None:
```
Comment (extsoft) [inline oct/tests/api/registration_pattern.py]: @AlekseyDidilov, please update method to the following
```suggestion
    def registration_response(self) -> str:
```

File: oct/tests/api/registration_pattern.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+            "lastname": self._full_name[1],
+            "email": self._email,
+            "telephone": self._telephone,
+            "password": self._password,
+            "confirm": self._password,
+            "newsletter": "0",
+            "agree": "1",
+        }
+        request = requests.post("http://localhost/index.php?route=account/register", parameters)
+        assert "Your Account Has Been Created!" in request.text
```
Comment (extsoft) [inline oct/tests/api/registration_pattern.py]: @AlekseyDidilov `assert` always must be in a test.

File: oct/tests/api/registration.py [hunk 6]
```diff
@@ -1,25 +1,15 @@
...
-            "password": "1234",
-            "confirm": "1234",
-            "newsletter": 0,
-            "agree": 1,
-        }
-        request = requests.post("http://localhost/index.php?route=account/register", parameters)
-        assert "Your Account Has Been Created!" in request.text
+    def registration_positive_test(self) -> None:
+        RegistrationAPI(
+            ["Alex", "Ivanov"], "newaded@gmail.com", "+38093245423", "12345678"
```
Comment (extsoft) [inline oct/tests/api/registration.py]: @AlekseyDidilov create an object `User` which will consist of `Identity` (first name, last name, phone) and `Credentials` (email and password). It is a bad idea to use a `list` object to represent first & last name.

Comment (AlekseyDidilov) [reply oct/tests/api/registration.py]: @extsoft all comments done.

File: oct/tests/api/registration_pattern.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    def get_newsletter(self) -> str:
+        return self._news_letter
+
+
+class User:
+    def __init__(self, identity: Identity, credentials: Credentials):
+        self._identity = identity
+        self._credentials = credentials
+
+    def registration_response(self) -> bool:
```
Comment (extsoft) [inline oct/tests/api/registration_pattern.py]: @AlekseyDidilov 
1. You return `bool` object, I would expect a response.
2. Is it a good place for this method?

Comment (AlekseyDidilov) [reply oct/tests/api/registration_pattern.py]: @extsoft Thank, now it returns response, and I renamed class name, which refers to registration_response method.

**Pull request merged**